File: src/problem211.py
from src.helpers.division import divisors
import math

def f(n):
    return sum(map(lambda x: x * x, divisors(n)))

def isSquare(n):
    return math.sqrt(n).is_integer()

def problem211():
    total = 0
    for i in range(1, 64_000_000):
        if i % 100_000 == 0:
            print(i, i / int(64e6) * 100)
        if isSquare(f(i)):
            total += i
    print(total)

# 1922364685
# 14:28:24.920231 ELAPSED - lol

File: src/problem211.cpp
#include <iostream>
#include <math.h>

long long sumDivSquares(long long number) {
    long long total = 0;
    int limit = (int) sqrt(number);
    for (long long i = 1; i <= limit; i++) {
        if (number % i == 0) {
            total += i * i;
            long long n = number / i;
            if (i != n) {
                total += n * n;
            }
        }
    }
    return total;
}

bool isPerfectSquare(long long number) {
    if (sqrt(number) == floor(sqrt(number))) {
        return true;
    }
    return false;
}

int main() {
    long long total = 0;
    for (long long i = 1; i < 64000000; i++) {
        if (isPerfectSquare(sumDivSquares(i))) {
            total += i;
        }
    }
    std::cout << total << std::endl;
    // 1922364685
    // real    63m3.445s
    return 0;
}
